file_browser/src/components/AddDIalog.tsx

```diff
@@ -0,0 +1,106 @@
+import { Dispatch, SetStateAction, useState } from "react";
+import { SystemItem } from "../classes/SystemItem";
+import { FileClass } from "../classes/File";
+import { Folder } from "../classes/Folder";
+import { observer } from "mobx-react-lite";
+import fileSystemStore from "../stores/FileSystemStore";
+
+const options = ["File", "Folder"];
+
+function AddDialog({
+  setDialog,
+}: {
+  setDialog: Dispatch<SetStateAction<boolean>>;
```

Comment:
A better name for this prop would be `onClose` and the type should be `() => void` as the dialog shouldn't care which specific state the parent uses or how it responds to closing. The parent will take care of setting the state.
Apply this to all dialogs.